src/main/java/com/pnu/dev/shpaltaif/controller/PostAdminController.java

```diff
@@ -0,0 +1,162 @@
+                          PostFiltersDto postFiltersDto) {
+
+        Page<Post> posts = postService.findAll(user, postFiltersDto, pageable);
+        List categories = categoryService.findAll();
+        List<PublicAccount> publicAccounts = publicAccountService.findAll();
+
+        model.addAttribute("pageable", pageable);
+        model.addAttribute("categories", categories);
+        model.addAttribute("publicAccounts", publicAccounts);
+        model.addAttribute("posts", posts);
+        model.addAttribute("postFilters", postFiltersDto);
+        return "admin/post/index";
+    }
+
+    @GetMapping("/{id}")
+    public String findById(@AuthenticationPrincipal User user, @PathVariable("id") Long id, Model model) {
+
+        Post post = postService.findById(user, id);
+
+        model.addAttribute("post", post);
+        return "admin/post/show";
+    }
+
+    @GetMapping("/new")
+    public String createForm(Model model) {
+
+        List categories = categoryService.findAll();
+        model.addAttribute("categories", categories);
+        return "admin/post/form";
+    }
+
+    @PostMapping("/new")
+    public String create(@AuthenticationPrincipal User user, @Validated PostDto postDto, RedirectAttributes redirectAttributes) {
+
+        redirectAttributes.addFlashAttribute(FLASH_MESSAGE_SUCCESS, "Пост успішно створено!");
+
+        postService.create(user, postDto);
+        return "redirect:/admin/posts";
+    }
+
+    @GetMapping("/edit/{id}")
+    public String editForm(@AuthenticationPrincipal User user, @PathVariable("id") Long id, Model model) {
+
+        Post post = postService.findById(user, id);
+        List categories = categoryService.findAll();
+        model.addAttribute("categories", categories);
+        model.addAttribute("post", post);
+
+        return "admin/post/form";
+    }
+
+    @PostMapping("/update/{id}")
+    public String update(@AuthenticationPrincipal User user, @PathVariable("id") Long id, @Validated PostDto postDto,
+                         RedirectAttributes redirectAttributes) {
+
+        System.out.println(postDto.toString());
+        postService.update(user, id, postDto);
+
+        redirectAttributes.addFlashAttribute(FLASH_MESSAGE_SUCCESS, "Пост успішно оновлено!");
+
+        return "redirect:/admin/posts";
+    }
+
+    @GetMapping("/deactivate/{id}")
+    public String deactivateConfirmation(@PathVariable("id") Long id, Model model,
+                                         HttpServletRequest request) {
+
+        model.addAttribute("message", "Ви впевнені, що справді хочете перемістити пост в архів?");
+        model.addAttribute("returnBackUrl", HttpUtils.getPreviousPageUrl(request));
+        model.addAttribute("actionUrl", "/admin/posts/deactivate/" + id);
+
+        return "admin/common/deleteConfirmation";
+    }
+
+    @PostMapping("/deactivate/{id}")
+    public String deactivate(@AuthenticationPrincipal User user, @PathVariable("id") Long id,
```

Comment:
We also need ability to activate deactivated post

Reply:
There is a checkbox in edit post form "activate before saving"
![image](https://user-images.githubusercontent.com/33934184/94848299-7ef52800-042c-11eb-8a15-892aad155014.png)

Reply:
We also need ability to activate post from posts page (the same place, where we have the ability to deactivate posts).
We shouldn't force our users to perform such workarounds.